src/compositor.c

```diff
@@ -29,52 +29,120 @@
  * in that mode.
  */
 
+/**
+ * Terminology:
+ * Zico core    - Xtensa core, running DryOS, used for graphics acceeleration,
+ *                GUI rendering and compositing. Connected with GPU (Takumi
+ *                GV550 IP core)
+ * mzrm         - Marius - Zico remote messaging (?). RPC used to talk between
+ *                main ARM core (named Marius on D6) and Zico.
+ * Ximr         - Render MIXer (?). Low level compositor on D6 and up.
+ * Ximr Context - Data structure representing Ximr configuration, sent via mzrm
+ *                using XimrExe() to Zico for render.
+ * Input layer  - Layer (described by MARV struct) to be included in mix.
+ * XOC          - Ximr Output Chunk. Buffer where final image is rendered.
+ * XCM          - Ximr Context Maker. "Userspace" tool to deal with Ximr setup.
+ *                Exists on D8 and up, from EOS R (M50 uses pure Ximr)
+ */
+
 #include "dryos.h"
 #include "bmp.h"
 #include "compositor.h"
 
-extern int uart_printf(const char * fmt, ...);
-
-#ifdef FEATURE_COMPOSITOR_XCM
+#ifdef CONFIG_COMPOSITOR_DEDICATED_LAYER
+/**
+ * refreshVrmsSurface (renamed to VMIX_TransferRectangleToVram on Digic X)
+ * is used by Canon code to render Input layers into XOC.
+ *
+ * On pure XIMR cameras it uses XIMR functions to setup Ximr Context
+ */
+extern void      refreshVrmsSurface();
+extern uint32_t  display_refresh_needed;
 
+// Store our Input Layer ID.
 int _rgb_vram_layer = 0;
```

Comment:
I guess the layer IDs should be consts, CANON_LAYER or something

Reply:
Discussed in discord, fine for now

Reply:
I addressed it in 26e9d97dc1573a79b51500d5c1c60719d4e7f6d5